src/gui/network_test.go

```diff
@@ -0,0 +1,292 @@
+package gui
+
+import (
+	"encoding/json"
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/skycoin/skycoin/src/daemon"
+)
+
+func TestConnection(t *testing.T) {
+	tt := []struct {
+		name                       string
+		method                     string
+		status                     int
+		err                        string
+		addr                       string
+		gatewayGetConnectionResult *daemon.Connection
+		result                     *daemon.Connection
+	}{
+		{
+			name:   "405",
+			method: http.MethodPost,
+			status: http.StatusMethodNotAllowed,
+			err:    "405 Method Not Allowed",
+		},
+		{
+			name:   "404 - empty addr",
+			method: http.MethodGet,
+			status: http.StatusNotFound,
+			err:    "404 Not Found",
+			addr:   "",
+			gatewayGetConnectionResult: nil,
+			result: nil,
+		},
+		{
+			name:   "200",
+			method: http.MethodGet,
+			status: http.StatusOK,
+			err:    "",
+			addr:   "addr",
+			gatewayGetConnectionResult: &daemon.Connection{},
+			result: &daemon.Connection{},
+		},
+	}
+	for _, tc := range tt {
+		t.Run(tc.name, func(t *testing.T) {
+			endpoint := "/network/connection"
+			gateway := NewGatewayerMock()
+			gateway.On("GetConnection", tc.addr).Return(tc.gatewayGetConnectionResult)
+			v := url.Values{}
+			if tc.addr != "" {
+				v.Add("addr", tc.addr)
+			}
+			if len(v) > 0 {
+				endpoint += "?" + v.Encode()
+			}
+			req, err := http.NewRequest(tc.method, endpoint, nil)
+			require.NoError(t, err)
+
+			rr := httptest.NewRecorder()
+			handler := NewServerMux(configuredHost, ".", gateway, &CSRFStore{})
+			handler.ServeHTTP(rr, req)
+
+			status := rr.Code
+			require.Equal(t, tc.status, status, "case: %s, handler returned wrong status code: got `%v` want `%v`", tc.name, status, tc.status)
+
+			if status != http.StatusOK {
+				require.Equal(t, tc.err, strings.TrimSpace(rr.Body.String()), "case: %s, handler returned wrong error message: got `%v`| %s, want `%v`",
+					tc.name, strings.TrimSpace(rr.Body.String()), status, tc.err)
```

Comment:
it should be 
`handler returned wrong error message: got `%v`| %d, want `%v`
`status` is an `int`

Reply:
@therealssj, fixed.